networkx/algorithms/shortest_paths/dense.py

```diff
@@ -218,6 +228,15 @@ def floyd_warshall(G, weight="weight"):
        A dictionary,  keyed by source and target, of shortest paths distances
        between nodes.
 
+    Examples
+    --------
+    >>> G = nx.DiGraph()
+    >>> G.add_weighted_edges_from([(0, 1, 5), (1, 2, 2), (2, 3, -3), (1, 3, 10),(3, 2, 8)])
+    >>> fw = nx.floyd_warshall(G, weight='weight')
+    >>> results = {a:dict(b) for a,b in fw.items()}
+    >>> print(results)
+    {0: {0: 0, 1: 5, 2: 7, 3: 4}, 1: {1: 0, 2: 2, 3: -1, 0: inf}, 2: {2: 0, 3: -3, 0: inf, 1: inf}, 3: {3: 0, 2: 8, 0: inf, 1: inf}}
```

Comment:
This seems unnecessary - can't we just output the results of `floyd_warshall` directly?

Reply:
We can, I just thought that maybe we could make it more comprehensible. When I was making this example, I found the output to be a little long because of the text and anyone using this function would probably like to remove it. So I thought to directly add this to the documentation. If anyone wants to use it as it is they can simply remove the fw. If you feel it is unnecessary I can remove it. Please let me know.
![image](https://user-images.githubusercontent.com/74042272/235416802-d7bdec2f-a00e-4ea8-8684-688e1bd0893a.png)

Reply:
Ah indeed, that's a very nasty repr due to the defaultdict + lambdas. I wonder if we can't improve the situation somewhat, but for now the proposed workaround seems good!